### PR TITLE
Emoji autocomplete

### LINKED ISSUE #2775: Add autocomplete to comment fields (starting with emoji) using the Horsey library
### Please describe the problem (or idea)

We have nice emoji now possible by using `\:smiley:` syntax. Let's make this autocomplete like it does in GitHub -- this will be a fun little project! 👍 😃 🐱 🐴 🌴 

image

https://bevacqua.github.io/horsey/

It should be pretty easy to implement! 

Follow-up ideas: 

1. autocompletion of names when you begin typing `@`
2. maybe even research notes when you start typing `#`

We'll probably need a string list of emoji names from somewhere to do this. Let's implement it in `app/assets/javascripts/comment-autocomplete.js` file!

### PR DESCRIPTION
**Pull request by ViditChitkara**
closes #2775

### CODE REVIEW

**Comment by namangupta01** [comment]
Hey @ViditChitkara, Can you please add a gif or image ?

File: app/views/comments/_edit.html.erb [hunk 1]
```diff
@@ -150,3 +149,15 @@
 </form>
 
 
+
+<style>
```
Comment (namangupta01) [inline app/views/comments/_edit.html.erb]: As these styles are also used in `_form.html.erb` we can write it in `app/assets/stylesheet/comment.css` ?

**Comment by ViditChitkara** [comment]
Hi @namangupta01 , this not fully complete till now there are some changes which need to be done. I would add review-me label once it is complete.

**Comment by ViditChitkara** [comment]
emoji-1
The  drop-down will be triggered by a ":"

**Comment by ViditChitkara** [comment]
Fixed some minor code tweaks. @jywarren is it good to go?

File: app/views/comments/_edit.html.erb (hunk 1 above)
Comment (ViditChitkara) [reply app/views/comments/_edit.html.erb]: done

File: app/views/comments/_form.html.erb [hunk 2]
```diff
@@ -64,11 +66,25 @@
   </script>
 
   <%= javascript_include_tag "dragdrop" %>
-  <script>    
+  <%= javascript_include_tag "horsey" %>
```
[the remark this answers is not in the document]
Comment (ViditChitkara) [reply app/views/comments/_form.html.erb]: sure, removed.

File: app/views/comments/_form.html.erb [hunk 3]
```diff
@@ -48,6 +48,8 @@
 
   <div class="well col-md-11" id="preview" style="background:white;display: none">
   </div>
+  <%= javascript_include_tag "horsey" %>
```
[the remark this answers is not in the document]
Comment (ViditChitkara) [reply app/views/comments/_form.html.erb]: removed

**Comment by jywarren** [comment]
If @namangupta01 approves his review, we can merge! This is very exciting 😄 🐴 👍 🐉

File: app/views/comments/_edit.html.erb (hunk 1 above)
Comment (namangupta01) [reply app/views/comments/_edit.html.erb]: we can also move this to `app/assets/stylesheet/comment.css`

**Comment by namangupta01** [comment]
Looking awesome to me.. just need one small change requested above.

File: app/views/comments/_edit.html.erb (hunk 1 above)
Comment (ViditChitkara) [reply app/views/comments/_edit.html.erb]: done!! I removed it as it was not needed there!!

**Comment by ViditChitkara** [comment]
I have made the requested changes, @jywarren it travis goes well, we can merge this!

**Comment by namangupta01** [comment]
Awesome!

**Comment by sagarpreet-chadha** [comment]
@ViditChitkara ...this is really awesome 😄 💛 !

**Comment by ViditChitkara** [comment]
Thank you @sagarpreet-chadha 😄 👍

**Pull request merged**